base/abstractarray.jl

```diff
@@ -487,8 +486,9 @@ _getindex(::LinearFast, A::AbstractArray, I::Int) = error("indexing not defined
 function _getindex(::LinearFast, A::AbstractArray, I::Real...)
     @_inline_meta
     # We must check bounds for sub2ind; so we can then call unsafe_getindex
-    checkbounds(A, I...)
-    unsafe_getindex(A, sub2ind(size(A), to_indexes(I...)...))
+    J = to_indexes(I...)
+    checkbounds(A, J...)
```

Comment:
Yup, but this is scalar indexing so `I<:Tuple{Vararg{Real}}`. It's something that I think we can get rid of once float indices are no longer supported, but that will take some thought about what to do with non-`Int` integers.  We'll still want to convert small unsigned integers to `Int` before doing sub2ind arithmetic... and ideally also allow larger integers, too.